### PR TITLE
Default configurations

### PR DESCRIPTION
**Pull request by kkaempf**
First piece of @dmacvicar's pull request. Rebased to current master, split to smaller pieces

### CODE REVIEW

File: lib/nailed/config.rb [hunk 1]
```diff
@@ -3,13 +3,23 @@
 #
 module Nailed
 
+  DEFAULT_CONFIG_PATH =
+    File.join(TOPLEVEL, "nailed", "default-config.yml")
```
Comment (MaximilianMeister) [inline lib/nailed/config.rb]: this should be in the `config` directory as well

**Comment by MaximilianMeister** [comment]
please keep the default configs, the yml files also in the config folder. right now they somehow moved to the lib folder, where they don't belong :)

**Comment by kkaempf** [comment]
Rebased to master. Default config is in config/ now.

File: lib/nailed/config.rb [hunk 2]
```diff
@@ -3,13 +3,23 @@
...
+  CONFIG_PATH =
+    File.join(TOPLEVEL, "config", "config.yml")
   #
   # Config
   #
   class Config
     # load config.yml once
     def self.content
-      @@conf ||= File.join(TOPLEVEL,"config","config.yml")
+      unless @@conf
```
Comment (MaximilianMeister) [inline lib/nailed/config.rb]: removing the memoization actually breaks it here :/ when `@@conf` doesn't exist yet we are running into an error.
I would suggest sth like
```ruby
@@conf ||= DEFAULT_CONFIG_PATH
if File.exist?(CONFIG_PATH)
  @@conf = CONFIG_PATH
end
```

File: lib/nailed/default-config.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (MaximilianMeister) [inline lib/nailed/default-config.yml]: this file and the default-colors.yml are still in the wrong place. they should go to the config folder

**Comment by MaximilianMeister** [comment]
overall I am not yet convinced by this. The mozilla default config is nice as an example but do we really need it?

I gave it a run and first it took ages to collect the bugs, and in the end they did't even end up in the database. I haven't further debugged this, maybe it helps choosing another project, but overall it doesn't work as expected.

**Comment by kkaempf** [comment]
Never mind.
I meanwhile made so many changes to have it perfect for my team that its probably too hard to merge it with your team's requirements anyways.

**Pull request closed without merging**